ssl/statem/statem_clnt.c

```diff
@@ -830,6 +830,7 @@ int tls_construct_client_hello(SSL *s)
         (sess->not_resumable)) {
         if (!ssl_get_new_session(s, 0))
             goto err;
+        s->session->ssl_version = s->version;
```

Comment:
This doesn't do anything. `ssl_get_new_session` always initializes it to `s->version`.